### ISSUE TITLE
Support for result export to JSON file

### ISSUE THREAD

**Issue by souvikinator** [DarthCucumber/unwee#2]
Currently the result is exported  in Text file when `-o` flag is used. Add support for  export to JSON file

**Comment by chinmaym07** [comment]
I explored the project . I tried to run on sample url's in the test file. I have gone though the code .
I want to try this .. if it's available . I am totally new to `go`, but I am aware of the syntax.  If you can please help me give a quick kickstart that will be really helpful to me .

**Comment by souvikinator** [comment]
@chinmaym07 Hey, thanks for reaching out. So if you have gone through the code you must have noticed a `TODO:` in **main.go** `line: 55`. `-o` output flag takes file name as input and if nothing is provided file name is generated based on time and date. You might want take a look at that.

You can probably make another `boolean` flag `-j` which when used will write to `JSON` file else default will be `TXT` file. Also in `utils.go` you can see function `WriteToFile` which writes to plain text file. You may rename it to `WriteToTextFile` (optional) and create a new function `WriteToJsonFile` which will be called when `-j` flag is used otherwise `WriteToTextFile` will be called. 

Lemme know if you face any issues.

**Comment by chinmaym07** [comment]
I was following the same approach. Will soon create a PR .thankyou ...

**Comment by chinmaym07** [comment]
Hey I have defined the function `WriteToJsonFile` in Utils  file & it works fine I think (I tested in place of `WriteToFile`)..The only issue is I am unable to find the type of output file from the flags whether it should be `.json` or `.txt` file as we are creating the file name if the file is not present . Please help . !
I think we should use one flag for file name/path & one for type of file .. 

https://user-images.githubusercontent.com/44670961/136626559-386d4291-167c-4e70-bee8-799b4882d9ac.mp4

**Comment by souvikinator** [comment]
@chinmaym07 Hey great work, that looks good. (**make sure to remove the extra output in the console**)
I see you are currently passing file name to `-j` flag. `-o` performs similar functionality, it takes file name as input. So one thing you can do is, make `-j` a boolean flag and make a variable `var fileExt string = ".txt"` (default value is **txt**) now when someone uses `-j` flag you you just set `fileExt=".json"` and concat the `fileExt` variable with the name of the file that is being generated (i.e when no file name is provided using `-o` flag) and if the user provides file name (using `-o`) then no need to concat `fileExt` with the file name.

Let me know if you don't understand anything.

**Comment by chinmaym07** [comment]
Yeah sure .. I just did that(output in console) for debugging purpose & will remove once the final testing is done . I think the step to generate a random file in `output` directory if the file name is not provided is not supported right now .

![image](https://user-images.githubusercontent.com/44670961/136647099-b5cdf65e-b01b-46be-81b5-a0a24bce1caf.png)

Will this work ?

https://user-images.githubusercontent.com/44670961/136647542-dd15669f-8fde-4406-95d0-f932d3df18df.mp4

**Comment by souvikinator** [comment]
> I think the step to generate a random file in `output` directory if the file name is not provided is not supported right now .

Well if you look at the `else if ` statement at line number 85 in `main.go`  in the video you attached in previous comment :

https://user-images.githubusercontent.com/44670961/136626559-386d4291-167c-4e70-bee8-799b4882d9ac.mp4

 you can see it generates file based on **current date and time** 

inside that just check if `-j` flag is used or not. If used then just change the file extension from `.txt` to `.json` and if user wants to provide output file then they can use `-o abc.txt` or `-o abc.json`
Currently `-j` flag is string type i.e it takes string as value change it to boolean so if `-j` is used it'll be `true` else `false`

**Comment by chinmaym07** [comment]
Yup I got your point that's why I asked you to use one flag for file & one flag for type of file . I will rectify it & will create a pr soon by today.
